src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -1,39 +1,194 @@
+      .add(StructField("expirationDate", StringType, false))
+    // Create a string with specified bytes
+    val byteString = "a" * payloadBytes
+
+    // Maximum number of records per partition
+    val maxPartitionRecords = 50000
+
+    // Calculating the number of partitions and the records per partition
+    val numPartitions       = math.ceil(numRecords.toDouble / maxPartitionRecords).toInt
+    val recordsPerPartition = math.ceil(numRecords.toDouble / numPartitions).toInt
+
+    val newdata = Range(1, recordsPerPartition).map(
+      x =>
+        Seq(UUID.randomUUID().toString(),
+            byteString,
+            UUID.randomUUID().toString(),
+            tempDateGenerator()))
+    val rows = newdata.map(x => Row(x: _*))
+    val rdd  = spark.sparkContext.makeRDD(rows)
+    val df   = spark.createDataFrame(rdd, schema)
+
+    val dfs = Range(1, numPartitions).map(x => df)
+
+    // Union the Sequence of DataFrames onto finalTable
+    val duplicateDf: DataFrame = dfs.reduce((l, r) => l.union(r))
+
+    import spark.implicits._
+    // Create temporary dataset to make changes on
+    val tempDs: Dataset[LoadGenTemp] = duplicateDf.as[LoadGenTemp]
+
+    // Create random data
+    val idRandomDs: Dataset[LoadGenTemp] = tempDs.map {
+      case change => change.copy(id = UUID.randomUUID().toString)
+    }
+    val dimRandomDs: Dataset[LoadGenTemp] = idRandomDs.map {
+      case change => change.copy(dimensionid = UUID.randomUUID().toString)
+    }
+    val finalDs = dimRandomDs.map {
+      case change => change.copy(expirationdate = tempDateGenerator())
+    }
+
+    //Convert back to a DF and return
+    // Return
+    finalDs.toDF()
+  }
+
+  // Create date data with 2016-12-25, 2017-12-25, 2018-12-25
+  def tempDateGenerator(): String = {
+    val r = new scala.util.Random()
+    val c = 0 + r.nextInt(3)
+    c match {
+      case 0 => "2016-12-25"
+      case 1 => "2017-12-25"
+      case 2 => "2018-12-25"
+      case _ => "2222-22-22"
+    }
+  }
 
   /**
     * Generates a test dataframe with keys from the parent used as foreign keys
     * @param numRecords Number of records in the dataframe
     * @param payloadBytes Payload size
     * @return The test dataframe
     */
-  def generateTableFromParent(numRecords: Int, payloadBytes: Int, parent: DataFrame): DataFrame = ???
+  def generateTableFromParent(spark: SparkSession,
+                              numRecords: Int,
+                              payloadBytes: Int,
+                              parentNumRecords: Int,
+                              parent: DataFrame): DataFrame = {
+
+    val byteString = "a" * payloadBytes
+
+    // Create a temporary dimension dataframe with the incorrect column names
+    val oldDimensionDf = parent
+      .select("dimensionid")
+      .limit(numRecords)
+      .withColumn("payload", lit(byteString))
+      .withColumn("subdimensionid", lit(UUID.randomUUID().toString()))
+      .withColumn("expirationDate", lit("2222-22-22"))
```

Comment:
hardcoded TODO?

Reply:
I have "2222-22-22" as a dummy data because I thought the whole point was that the dimension tables I'm generating don't want an expiration date column, and thus I just put in a large date that won't ever be expired.